docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -235,6 +235,16 @@ Support for non-ref-forwarding class components in the `component` prop or as im
 Otherwise check out the [Caveat with refs](/guides/composition/#caveat-with-refs) section in the composition guide to find out how to migrate.
 This change affects almost all components where you're using the `component` prop or passing `children` to components that require `children` to be elements (e.g. `<MenuList><CustomMenuItem /></MenuList>`)
 
+### Ref type specificity
```

Comment:
Is this required for all components? I thought it is specific to the `Card` component. Can you double check?

Reply:
You are right. This is only for `Card` component so far but we may have other components with the same behaviour, so I modified it to "some components may have ...". Check out https://github.com/mui-org/material-ui/pull/30114/commits/d58ff3a4ebdc971f75fe83c69295647b49955cde

Reply:
I didn't say "Card component requires `HTMLDivElement`..." because I see [here](https://codesandbox.io/s/basiccard-material-demo-forked-qsfzn?file=/demo.tsx)that `HTMLInputElement` also keeps the ts error quiet.

Reply:
We need to see what exactly us the breaking change. Migration is hard, so saying things like: some component may require something is not helpful.

Reply:
@mnajdova Ah, I see :) I went through the repo and found those that expect a specific ref type and enumerated them in the doc. Check out https://github.com/mui-org/material-ui/pull/30114/commits/30d03cfc37abd686908c8724ad73b47a3aa978a7